### PR TITLE
Fixed Delete Employees

### PR DESCRIPTION
**Pull request by RohitValiveti**
### Summary 

Prior to these changes, when trying to delete an employee on the admin side, the networks requests would just use the ConfirmationModal Component that was designed to only delete rider users. So this PR adjusted the ConfirmationModal component to account for if an employee is being deleted (i.e. an admin or a driver), and calls the proper Delete request.

### Test Plan 

Manual UI testing

### Notes 

Currently, in our user schema, if someone is a driver and an admin, they are treated as separate objects in the driver db and the admin db with different ids -- there is not relation between them besides the admin attribute in the driver object being set to true. Because of this, if someone who is an admin and a driver is being deleted, I only deleted their driver user and kept their admin user.

Also, if an admin is deleted who is also a driver, their 'admin' attribute in their driver db object needs to be updated to false, but the difficulty comes in retrieving the "driver" id. This is why I think there may need to be a future schema change for our db that links 2 users who are a driver and an admin. Perhaps some sort of association table may need to be implemented.

[Trello Card](https://trello.com/c/PJx5Hhyh/445-delete-employees-is-currently-not-working)

### CODE REVIEW

File: frontend/src/components/Modal/ConfirmationModal.tsx [hunk 1]
```diff
@@ -25,18 +35,63 @@ const ConfirmationModal = ({ open, rider, onClose }: ConfirmationProps) => {
   };
 
   const studentDelete = () => {
```
Comment (pratyush1712) [inline frontend/src/components/Modal/ConfirmationModal.tsx:37]: I think the function should be renamed. This Modal is now used to delete both students and employees. So, naming it only studentDelete is confusing. I think you should create two functions: `employeeDelete` and `studentDelete`, each one getting called conditionally.

File: frontend/src/components/Modal/ConfirmationModal.tsx [hunk 2]
```diff
@@ -7,15 +7,25 @@ import { useRiders } from '../../context/RidersContext';
 import { Rider } from '../../types/index';
 import { useHistory } from 'react-router-dom';
 import { ToastStatus, useToast } from '../../context/toastContext';
+import { useEmployees } from '../../context/EmployeesContext';
 
 type ConfirmationProps = {
   open: boolean;
   rider?: Rider;
   onClose: () => void;
+  employeeId?: string;
```
Comment (pratyush1712) [inline frontend/src/components/Modal/ConfirmationModal.tsx:16]: I think the Props should be more descriptive. For example, 
`open, isRider: bool, onClose, id, role` (id is employee id or student id depending on the isRider bool) or
`open, isRider: bool, onClose, user` (here user is Admin | Driver type or Rider Type depending on isRider)

I think we are only using the rider.id, so we don't need the  entire rider object

**Review by pratyush1712 — COMMENTED** [review]
Good job! The delete option is working on my end. Thank you so much for fixing the bug and pointing out the issues in the schema. We should totally discuss them in our next work session. For now, I am leaving some comments on your PR.

File: frontend/src/components/Modal/ConfirmationModal.tsx [hunk 3]
```diff
@@ -25,18 +35,63 @@ const ConfirmationModal = ({ open, rider, onClose }: ConfirmationProps) => {
...
+      fetch(
+        `/api/drivers/${employeeId ? employeeId : ''}`,
+        withDefaults({
+          method: 'DELETE',
+        })
+      )
+        .then(refreshDrivers)
+        .then(refreshAdmins)
+        .then(() => {
+          history.push('/admins');
```
[the remark this answers is not in the document]
Comment (RohitValiveti) [reply frontend/src/components/Modal/ConfirmationModal.tsx:77]: I think for the history.push(), it needs to remain as /admins. I tried what you said, but the app crashes after I delete an admin if I put /admins/employees.

File: frontend/src/components/Modal/ConfirmationModal.tsx [hunk 4]
```diff
@@ -25,18 +35,63 @@ const ConfirmationModal = ({ open, rider, onClose }: ConfirmationProps) => {
...
+        .then(() => {
+          history.push('/drivers');
+          showToast('The driver has been deleted.', ToastStatus.SUCCESS);
+          closeModal();
+        });
+    }
+    // Update both driver and admin dbs
+    else if (role === 'both') {
+      fetch(
+        `/api/drivers/${employeeId ? employeeId : ''}`,
```
[the remark this answers is not in the document]
Comment (RohitValiveti) [reply frontend/src/components/Modal/ConfirmationModal.tsx]: This is where the problem with the dbs comes in. The id of the user is not guaranteed to be the same in the driver and admin db. If a user is both an admin and driver, the id passed through is their driver id, so to delete the same user from  the admin db, there would need to be some way to retrieve their admin id as well. I commented this in the code as well for reference. Any suggestions? Or should we have some task that addresses this db fix

File: frontend/src/components/Modal/ConfirmationModal.tsx [hunk 5]
```diff
@@ -24,26 +32,56 @@ const ConfirmationModal = ({ open, rider, onClose }: ConfirmationProps) => {
...
       withDefaults({
         method: 'DELETE',
       })
     )
-      .then(refreshRiders)
+      .then(refreshFunc)
       .then(() => {
-        history.push('/riders');
-        showToast('The student has been deleted.', ToastStatus.SUCCESS);
+        history.push(
```
Comment (pratyush1712) [inline frontend/src/components/Modal/ConfirmationModal.tsx:50]: I think `/admin` is already present in the path, and we only need to redirect to `riders` or `employees` page.
This seems to work perfectly:
history.push(`/${userType === 'rider' ? 'riders' : 'employees'}`);

File: frontend/src/components/Modal/ConfirmationModal.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (pratyush1712) [reply frontend/src/components/Modal/ConfirmationModal.tsx]: hmm I looked into the way employees of type `both` are added. It seems like if we select both driver and admin then only a driver object is created and the admin field is set to true for that object. So, I don't think we need to make a separate API call to delete admin as well. Will look into it more in our next work session

**Review by pratyush1712 — APPROVED** [review]
Awesome work! I will create a Trello task for schema changes. For now, these changes are good to merge!

**Pull request merged**